File: src/main/java/teammates/common/datatransfer/attributes/Gender.java
package teammates.common.datatransfer.attributes;

import com.google.gson.annotations.SerializedName;

/**
 * Represents the gender of a student.
 */
public enum Gender {
    @SerializedName("male")
    MALE,
    @SerializedName("female")
    FEMALE,
    @SerializedName("other")
    OTHER,
    ;

    /**
     * Returns the Gender enum value corresponding to {@code gender} or OTHER by default.
     */
    public static Gender getGenderEnumValue(String gender) {
        try {
            return Gender.valueOf(gender.toUpperCase());
        } catch (IllegalArgumentException e) {
            return Gender.OTHER;
        }
    }
}

File: src/web/types/gender.ts
/**
 * Represents the gender of a student.
 */
export enum Gender {

  /**
   * Male gender.
   */
  MALE,

  /**
   * Female gender.
   */
  FEMALE,

  /**
   * Not specified gender.
   */
  OTHER,
}

export namespace Gender {

  /**
   * Returns an array containing the values of Gender enum.
   */
  export const enumValues: () => Gender[] = (): Gender[] => {
    const genders: Gender[] = [];
    genders.push(Gender.MALE);
    genders.push(Gender.FEMALE);
    genders.push(Gender.OTHER);
    return genders;
  };
}
